openfermioncirq/variational/objective.py

```diff
@@ -0,0 +1,59 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+from typing import Optional, Tuple, Union
+
+import numpy
+
+import cirq
+from cirq import abc
+
+
+class VariationalObjective(metaclass=abc.ABCMeta):
```

Comment:
Seriously, get in the habit of profuse documentation when you do things like this. I don't know what a metaclass is and you can be certain that neither do 99% of our users. I want to see a lot more documentation explaining why you are using advanced features of the language and what benefit it provides and what it even means. This will make the code much more readable to people who aren't familiar with, e.g. metaclasses.

Reply:
You don't need to know what a metaclass is; you just need to know what an abstract base class is and that this is how one uses them in Python. Anyway, let's talk about it tomorrow.